### ISSUE TITLE
Combobox search input not focusable within Drawer (Svelte 5)

### ISSUE THREAD

**Issue by chanmathew** [huntabyte/shadcn-svelte#1493]
### Describe the bug

Hi folks - Please see the below video. 

https://github.com/user-attachments/assets/c37ffd5b-911b-4f48-98b3-3ae35d94467e

There are a few issues:
1. When I have a combobox within a Drawer, I can't seem to select the search input at all
2. When you select anything, it will immediately close the drawer, which should not be the expected behavior as you may have a form and you need to complete the rest of the inputs

### Reproduction

Here's a repro: 
https://stackblitz.com/edit/github-aibhpm?file=src%2Froutes%2F%2Bpage.svelte

### System Info

```bash
System:
    OS: Linux 5.0 undefined
    CPU: (8) x64 Intel(R) Core(TM) i9-9880H CPU @ 2.30GHz
    Memory: 0 Bytes / 0 Bytes
    Shell: 1.0 - /bin/jsh
  Binaries:
    Node: 18.20.3 - /usr/local/bin/node
    Yarn: 1.22.19 - /usr/local/bin/yarn
    npm: 10.2.3 - /usr/local/bin/npm
    pnpm: 8.15.6 - /usr/local/bin/pnpm
  npmPackages:
    @sveltejs/kit: ^2.8.1 => 2.8.1 
    bits-ui: 1.0.0-next.57 => 1.0.0-next.57 
    cmdk-sv: ^0.0.18 => 0.0.18 
    formsnap: ^0.5.0 => 0.5.0 
    lucide-svelte: ^0.459.0 => 0.459.0 
    mode-watcher: ^0.2.1 => 0.2.1 
    svelte: ^5.2.0 => 5.2.0 
    svelte-sonner: ^0.3.28 => 0.3.28 
    sveltekit-superforms: ^2.20.1 => 2.20.1 
    vaul-svelte: ^0.3.2 => 0.3.2
```

### Severity

annoyance

**Comment by huntabyte** [comment]
You should be using `vaul-svelte@next` and shouldn't be using `cmdk-sv` should be using `Command` from `bits-ui@next` instead 😃

**Comment by chanmathew** [comment]
@huntabyte Thanks that seems to work!

**Comment by chanmathew** [comment]
Ah! actually i realized what the issue is @huntabyte - if you have snapPoints set on the drawer, then the input won't focus:

https://stackblitz.com/edit/github-aibhpm?file=src%2Froutes%2F%2Bpage.svelte

**Comment by huntabyte** [comment]
It seems to be focusing for me: 

https://github.com/user-attachments/assets/fb9f0899-bd9b-40ff-88d2-0c76f303a1bb

**Comment by chanmathew** [comment]
@huntabyte the combobox does open but if you have snapPoints set on the drawer, you can't focus the `input`:

![CleanShot 2024-11-16 at 12 22 12@2x](https://github.com/user-attachments/assets/af041def-ee28-472b-be65-f950b1aee671)

**Comment by huntabyte** [comment]
Ah the input _inside_ the popover, got it. And this _only_ happens when snap points are enabled? That is very strange as the two _should_ be completely unrelated.

**Comment by huntabyte** [comment]
Hm this seems more like a focus trap race condition going on here which would be something to do with Bits UI where focus remains trapped inside the Drawer vs moving into the Popover

**Comment by chanmathew** [comment]
Yup that's right! At least so far from my testing, it's only when the snapPoints is set.

**Comment by huntabyte** [comment]
Yeah, I see that in the reproduction... very interesting. I'll look into this further as I'm not sure if it falls on Bits UI or Vaul Svelte (though a vaul svelte action triggers this behavior, it could be an underlying issue with bits that we need to support)
